third_party/intel/lib/TritonRaiseBlockPointer/TritonRaiseBlockPointer.cpp

```diff
@@ -489,6 +489,235 @@ static llvm::raw_ostream &operator<<(llvm::raw_ostream &os,
 }
 #endif
 
+// Utility class aggregating information required to create a versioning
+// condition.
+class VersioningCondition {
+public:
+  VersioningCondition(Value S, Value BS) : S(S), BS(BS) {
+    assert(isValid() && "Invalid values supplied");
+  }
+  //  VersioningCondition() : S(), BS(){};
+  //  VersioningCondition(const VersioningCondition &other) {
+  //    S = other.S;
+  //    BS = other.BS;
+  //  }
+
+  // Create the condition: (S % BS == 0 && S > BS)
+  Value materialize(OpBuilder &builder, Location loc) const {
+    assert(S && BS && "Expecting valid values");
+    Value zero =
+        builder.createOrFold<arith::ConstantIntOp>(loc, 0, S.getType());
+    Value cmp1 = builder.create<arith::CmpIOp>(
+        loc, arith::CmpIPredicate::eq,
+        builder.create<arith::RemSIOp>(loc, S, BS), zero);
+    Value cmp2 =
+        builder.create<arith::CmpIOp>(loc, arith::CmpIPredicate::sgt, S, BS);
+    return builder.create<arith::AndIOp>(loc, cmp1, cmp2);
+  }
+
+private:
+  bool isValid() const {
+    Type SType = S.getType(), BSType = BS.getType();
+    if (!isa<IntegerType>(SType) || !isa<IntegerType>(BSType))
```

Comment:
wonder if something like below would work
```suggestion
    if (!isa<IntegerType>(SType, BSType))
```

Reply:
We can't do that. We can pass only on value (but we can pass a variadic list of template parameters).